apis/v1beta1/opentelemetrycollector_types.go

```diff
@@ -94,6 +94,12 @@ type OpenTelemetryCollectorSpec struct {
 	// +required
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Config Config `json:"config"`
+	// ConfigVersions defines the number versions to keep for the collector config. Each config version is stored in a separate ConfigMap.
+	// Defaults to 3. The minimum value is 1.
+	// +optional
+	// +kubebuilder:default:=3
+	// +kubebuilder:validation:Minimum:=1
+	ConfigVersions int `json:"configVersions,omitempty"`
```

Comment:
@swiatekm-sumo do you think we should also add a validation for configVersions in the webhook to ensure that it's always >1 or do we trust kubebuilder enough here?

Reply:
I trust OpenAPI validation, which is what it comes down to. In general, I'd like to rely more on these validations and less on the webhook, given that the latter can be disabled.